client/src/client/startup.py

```diff
@@ -94,44 +94,29 @@ def get_key(self, passphrase: str = "") -> bytes:
             (self._hardware_fingerprint() + passphrase).encode("utf-8")
         ).digest()
 
-    def generate_key_pair(self, password: str) -> KeyPair:
-        """Run OpenSSL to generate a pair of RSA keys.
+    def generate_key_pair(self, password: bytes) -> KeyPair:
+        """Run OpenSSL to generate a pair of EC keys.
 
         Save generated values to the .cans/keys directory.
         """
-        priv_key = str(
-            subprocess.run(
-                "openssl genrsa 4096",
-                capture_output=True,
-                text=True,
-                shell=True,
-            ).stdout
-        )
+        # TODO: Add private key encryption
+        password = password
 
-        with open(self.user_private_key_path, "w") as private_key_file:
-            private_key_file.write(priv_key)
-
-        pub_key = str(
-            subprocess.run(
-                "openssl rsa -in "
-                + self.user_private_key_path.as_posix()
-                + " -pubout",
-                capture_output=True,
-                text=True,
-                shell=True,
-            ).stdout
+        subprocess.run(
```

Comment:
Do we want to do it using subprocess? I know it's more of a general question but we're using pycryptodome anyway, we can generate keys there

Reply:
Truth be told we can also use jks instead of encrypting the key